_posts/2023-08-17-Announcing-Data-Prepper-2.4.0.md

```diff
@@ -0,0 +1,285 @@
+Scan can be configured with multiple buckets and start time, end time can be configured at bucket level which will override top level options. 
+User can also configure include, exclude lists at the bucket level to filter the data being scanned. 
+User can also configure pipeline with out start and end time and pipeline will start one-time scan to ingest all the existing data in the S3 buckets.
+
+The following source configuration can be used to get started with S3 scan.
+
+```
+source:
+  s3:
+    acknowledgments: true
+    scan:
+      start_time: 2023-01-01T00:00:00
+      end_time: 2023-12-31T23:59:59
+      buckets:
+      - bucket:
+          # start_time: 2023-01-01T00:00:00
+          # end_time: 2023-12-31T23:59:59
+          name: "s3-scan-bucket"
+          filter:
+            exclude_suffix:
+            - "*.log"
+            include_prefix:
+            - "prefix1/"
+    delete_s3_objects_on_read: true
+    codec:
+      newline:
+    aws:
+      region: "us-east-1"
+       sts_role_arn: "arn:aws:iam::1234567890:role/scan-role"
+```
+
+Objects scanned using S3 scan can be deleted after processed successfully using the following configuration in `s3` source. 
+This feature will delete S3 objects only when acknowledgements are enabled.
+
+```
+s3:
+  delete_s3_objects_on_read: true
+```
+
+Optionally, scanning the bucket can be configured to be done periodically on a user defined scheduled. 
+The following configuration can be used to schedule the scan.
+
+```
+source:
+  s3:
+    scan:
+      buckets:
+      - bucket:
+        name: "s3-scan-test"
+        filter:
+          exclude_suffix:
+          - "*.log"
+          include_prefix:
+          - "new"
+    scheduling:
+      interval: PT12H #Scan every 12 hours
+      count: 10       #Run 10 scans
+```
+
+See the documentation for complete set of [configurable options](https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sources/s3/).
+
+
+## Filtering in sinks
+
+User can configure the option of “include_keys” and “exclude_keys” under sink to allow or deny data being written to sinks. 
+This provides flexibility to ingest data from any source, apply common enrichments using a processor chain and selective send data to a specific sink like OpenSearch and send all other data to sink like S3 for archival purpose. 
+Archived data in S3 can be written in Parquet so it can be easily be queried by any other compatible engine. 
+Users can also then leverage S3 batch processing to hydrate archived data back to OpenSearch if needed for richer analytics. 
+Such filtering enables you to optimize your storage and reduce costs by sending specific data to specific sinks.
+
+```
+sink:
+- opensearch:
+  include_keys:
+  - srcport
+  - dstport
+  - srcaddr
+  - dstaddr
+  - start
+  - end
+
+- s3:
+  exclude_keys:
+  - version
+  - interfaceId
+```
+
+
+S3 sink codecs
```

Comment:
Rewrite for this section:

--- 

Data Prepper 2.4.0 adds new codecs to the [S3 sink](https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/sinks/s3/):

- **JSON codec**: The JSON codec writes an S3 objects as a valid JSON object and includes all events sent to S3 in the JSON array. This is useful when integrating with systems that expect JSON objects.
- **Arvo codec**: The Arvo codecs gives you the ability to define Arvo schemas and write events in the Arvo format.
- **Parquet codec**: The [Apache Parquet (Parquet)](https://parquet.apache.org/) codec lets you use the Parquet columnar format. By saving you data in the Parquet format, you efficiently retrieve that data at a later time for analysis. You can also define your schema in the Avro format inside the Parquet codec.
- **CSV codec**: The CSV codec allows you to write events in a delimited format, such as TSV or CSV.

The following example configures an Avro codec with a schema for network traffic:

```json
s3:
  codec:
    avro:
      schema: >
        {
          "type" : "record",
          "namespace" : "org.opensearch.dataprepper.examples",
          "name" : "NetworkTraffic",
          "fields" : [
            { "name" : "sourcePort", "type": "int"},
            { "name" : "destinationPort", "type": "int"},
            { "name" : "sourceAddress", "type" : "string"},
            { "name" : "destinationAddress", "type" : "string"},
            { "name" : "bytes", "type": "int"},
          ]
        }
```

Additionally, the S3 sink can now write compressed `gzip` or compressed `snappy` files to reduce your network or storage needs. To use this setting, add the following to your pipeline:

```json
s3:
  compression: gzip
```